0_config/src/wqp_utils.R

```diff
@@ -0,0 +1,94 @@
+inventory_wqp <- function(wqp_needs, wqp_variables) {
+  
+  # identify constituents we need
+  constituents <- as.character(unique(wqp_needs$variable))
+  
+  
+  wqp_args <- list(
+    characteristicName=NA, # to be updated each time through loop
+    siteid=NA #updated for each constituent
+  )
+  
+  # loop over the constituents, getting rows for each
+  sample_time <- system.time({
+    samples <- bind_rows(lapply(constituents, function(constituent) {
+      message(Sys.time(), ': getting inventory for ', constituent)
+      wqp_args$characteristicName <- wqp_variables$characteristicName[[constituent]]
+      wqp_args$siteid <- wqp_needs$site[wqp_needs$variable %in% constituent]
+      tryCatch({
+        wqp_wdat <- do.call(whatWQPdata, wqp_args)
+        mutate(wqp_wdat, constituent=constituent)
+      }, error=function(e) {
+        # keep going IFF the only error was that there weren't any matching sites
+        if(grepl('arguments imply differing number of rows', e$message)) {
+          NULL
+        } else {
+          stop(e)
+        }
+      })
+    }))
+  })
+  message(sprintf('sample inventory complete, required %0.0f seconds', sample_time[['elapsed']]))
+  
+  return(samples)
+}
+
+partition_wqp_inventory <- function(wqp_inventory, wqp_nrecords_chunk) {
```

Comment:
I know I didn't do a good job of this on the other repo, but can we add
`#' @param {paramname} {info here}` for each argument in the function? Will help us in the future to know what the expectations/assumptions are for those values

Reply:
documentation to all functions added